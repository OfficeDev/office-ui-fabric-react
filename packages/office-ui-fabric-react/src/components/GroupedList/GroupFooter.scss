@import '../../common/common';

.ms-groupFooter {
  position: relative;
  padding: 10px 84px;
  cursor: pointer;

  .ms-Link {
    font-size: $ms-font-size-s;
  }
}