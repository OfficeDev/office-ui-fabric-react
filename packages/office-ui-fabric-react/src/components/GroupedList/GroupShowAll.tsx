import { styled } from '../../Utilities';
import { getStyles } from './GroupShowAll.styles';
import { GroupShowAllBase } from './GroupShowAll.base';
import { IGroupShowAllProps } from './GroupShowAll.types';
export { IGroupShowAllProps };

export const GroupShowAll = styled(GroupShowAllBase, getStyles, undefined, { scope: 'GroupShowAll' });
