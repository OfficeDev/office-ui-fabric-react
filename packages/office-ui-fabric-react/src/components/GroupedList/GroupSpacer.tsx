import * as React from 'react';
import {
  css
} from '../../Utilities';
import * as stylesImport from './GroupSpacer.scss';
const styles: any = stylesImport;

export interface IGroupSpacerProps {
  count: number;
  widthUnit?: number;
}

const SPACER_WIDTH = 36;

export const GroupSpacer = (props: IGroupSpacerProps) =>
  props.count > 0 && (
    <span
      className={ css(
        'ms-GroupSpacer',
        styles.root
      ) }
      style={ { width: props.count * getWidthUnit(props.widthUnit) } }
    />
  );

function getWidthUnit(widthUnit: number | undefined): number {
  return typeof widthUnit === 'number' ? widthUnit : SPACER_WIDTH;
}