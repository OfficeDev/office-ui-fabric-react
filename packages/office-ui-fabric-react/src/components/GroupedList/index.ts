export * from './GroupedList';
export * from './GroupedList.base';
export * from './GroupedList.types';
export * from './GroupHeader';
export * from './GroupHeader.base';
export * from './GroupHeader.types';
export * from './GroupFooter';
export * from './GroupFooter.base';
export * from './GroupFooter.types';
export * from './GroupShowAll';
export * from './GroupShowAll.base';
export * from './GroupShowAll.types';
export * from './GroupSpacer';
export * from './GroupSpacer.types';
