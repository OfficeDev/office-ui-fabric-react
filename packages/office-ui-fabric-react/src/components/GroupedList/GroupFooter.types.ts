import { IStyleFunctionOrObject } from '../../Utilities';
import { IGroupDividerProps } from './GroupedList.types';
import { ITheme, IStyle } from '../../Styling';

export interface IGroupFooterProps extends IGroupDividerProps {
  /**
   * Theme provided by context
   */
  theme?: ITheme;

  /**
   * Overridable styles
   */
  styles?: IStyleFunctionOrObject<IGroupFooterStyleProps, IGroupFooterStyles>;

  /**
   * Custom classname
   */
  className?: string;
}

export type IGroupFooterStyleProps = Required<Pick<IGroupFooterProps, 'theme'>> &
  Pick<IGroupFooterProps, 'selected' | 'className'> & {
    isCollapsed?: boolean;
  };

export interface IGroupFooterStyles {
  root: IStyle;
}
