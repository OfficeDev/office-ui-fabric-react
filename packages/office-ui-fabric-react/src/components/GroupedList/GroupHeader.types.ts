import { IStyleFunctionOrObject } from '../../Utilities';
import { IGroupDividerProps } from './GroupedList.types';
import { ITheme, IStyle } from '../../Styling';

export interface IGroupHeaderProps extends IGroupDividerProps {
  /**
   * Theme provided by context
   */
  theme?: ITheme;

  /**
   * Overridable styles
   */
  styles?: IStyleFunctionOrObject<IGroupHeaderStyleProps, IGroupHeaderStyles>;

  /**
   * Custom classname
   */
  className?: string;
}

export type IGroupHeaderStyleProps = Required<Pick<IGroupHeaderProps, 'theme'>> &
  Pick<IGroupHeaderProps, 'selected' | 'className'> & {
    isCollapsed?: boolean;
  };

export interface IGroupHeaderStyles {
  root: IStyle;
  groupHeaderContainer: IStyle;
  headerCount: IStyle;
  check: IStyle;
  dropIcon: IStyle;
  expand: IStyle;
  expandIsCollapsed: IStyle;
  title: IStyle;
}
