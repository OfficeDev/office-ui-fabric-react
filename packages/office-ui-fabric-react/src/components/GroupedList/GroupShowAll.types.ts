import { ITheme, IStyle } from '../../Styling';
import { IGroupDividerProps } from './GroupedList.types';
import { IStyleFunctionOrObject } from '../../Utilities';

export interface IGroupShowAllProps extends IGroupDividerProps {
  /**
   * Theme provided by context
   */
  theme?: ITheme;

  /**
   * Overridable styles
   */
  styles?: IStyleFunctionOrObject<IGroupShowAllStyleProps, IGroupShowAllStyles>;

  /**
   * The Show All link text.
   * @default 'Show All'
   */
  showAllLinkText?: string;
}

export interface IGroupShowAllStyleProps {
  theme: ITheme;
}

export interface IGroupShowAllStyles {
  root: IStyle;
}
