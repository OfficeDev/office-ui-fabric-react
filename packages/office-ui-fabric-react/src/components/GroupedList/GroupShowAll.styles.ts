import { IGroupShowAllStyleProps, IGroupShowAllStyles } from './GroupShowAll.types';
import { getGlobalClassNames, FontSizes } from '../../Styling';

const GlobalClassNames = {
  root: 'ms-GroupShowAll',
  link: 'ms-Link'
};

export const getStyles = (props: IGroupShowAllStyleProps): IGroupShowAllStyles => {
  const { theme } = props;
  const classNames = getGlobalClassNames(GlobalClassNames, theme);

  return {
    root: [
      classNames.root,
      {
        position: 'relative',
        padding: '10px 84px',
        cursor: 'pointer',
        selectors: {
          [`.${classNames.link}`]: {
            fontSize: FontSizes.small
          }
        }
      }
    ]
  };
};
