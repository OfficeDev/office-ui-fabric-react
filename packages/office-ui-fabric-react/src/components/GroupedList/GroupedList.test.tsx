import * as React from 'react';
import * as renderer from 'react-test-renderer';

import { GroupedList } from './GroupedList';
import { List } from 'office-ui-fabric-react/lib/components/List';

import { createGroups } from 'office-ui-fabric-react/lib/utilities/exampleData';

// Populate mock data for testing
function mockData(count: number): any {
  const data = [];
  let _data = {};

  for (let i = 0; i < count; i++) {
    _data = {
      key: i,
      name: 'Item ' + i,
      value: i
    };
    data.push(_data);
  }

  return data;
}

describe('GroupedList', () => {
  it('renders grouped list correctly', () => {
    List.prototype.componentDidMount = jest.fn();

    const component = renderer.create(
      // tslint:disable-next-line:jsx-no-lambda
      <GroupedList items={mockData(3)} onRenderCell={() => null} groups={createGroups(3, 3, 0, 3)} />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders grouped list with custom expand button aria label correctly', () => {
    List.prototype.componentDidMount = jest.fn();

    const component = renderer.create(
      <GroupedList
        items={mockData(3)}
        // tslint:disable-next-line:jsx-no-lambda
        onRenderCell={() => null}
        groups={createGroups(3, 3, 0, 3)}
        groupProps={{
          headerProps: {
            expandButtonProps: {
              'aria-label': 'Custom aria label'
            }
          }
        }}
      />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
