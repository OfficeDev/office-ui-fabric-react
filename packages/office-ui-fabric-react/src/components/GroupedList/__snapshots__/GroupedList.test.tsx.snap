// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupedList renders grouped list correctly 1`] = `
<div
  className=
      ms-GroupedList
      {
        -moz-osx-font-smoothing: grayscale;
        -webkit-font-smoothing: antialiased;
        font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        font-size: 12px;
        font-weight: 400;
        position: relative;
      }
      & :global(.ms-List-cell) {
        min-height: 36px;
      }
  data-automationid="GroupedList"
  data-is-scrollable="false"
  role="presentation"
>
  <div
    className="ms-List"
    role="list"
  >
    <div
      className="ms-List-surface"
      role="presentation"
    />
  </div>
</div>
`;

exports[`GroupedList renders grouped list with custom expand button aria label correctly 1`] = `
<div
  className=
      ms-GroupedList
      {
        -moz-osx-font-smoothing: grayscale;
        -webkit-font-smoothing: antialiased;
        font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        font-size: 12px;
        font-weight: 400;
        position: relative;
      }
      & :global(.ms-List-cell) {
        min-height: 36px;
      }
  data-automationid="GroupedList"
  data-is-scrollable="false"
  role="presentation"
>
  <div
    className="ms-List"
    role="list"
  >
    <div
      className="ms-List-surface"
      role="presentation"
    />
  </div>
</div>
`;
